src/temaWeek8Exception/main/Student.java

```diff
@@ -0,0 +1,160 @@
+package temaWeek8Exception.main;
+/**
+ * Class student with the following attributes:first name,
+ * last name, date of birth, gender and CNP.
+ */
+
+import java.time.Year;
+import java.util.Comparator;
+import java.util.Objects;
+
+public class Student {
```

Comment:
It feels like this class does too much. Maybe extract the `Comparator` to its own class.